utils/search/search_general.py

```diff
@@ -22,19 +22,320 @@
+    tag_weight = settings.DEFAULT_SEARCH_WEIGHTS['tag']
+    description_weight = settings.DEFAULT_SEARCH_WEIGHTS['description']
+    username_weight = settings.DEFAULT_SEARCH_WEIGHTS['username']
+    pack_tokenized_weight = settings.DEFAULT_SEARCH_WEIGHTS['pack_tokenized']
+    original_filename_weight = settings.DEFAULT_SEARCH_WEIGHTS['original_filename']
+
+    # Parse advanced search options
+    advanced = request.GET.get("advanced", "")
+    advanced_search_params_dict = {}
+
+    # if advanced search
+    if advanced == "1":
+        a_tag = request.GET.get("a_tag", "")
+        a_filename = request.GET.get("a_filename", "")
+        a_description = request.GET.get("a_description", "")
+        a_packname = request.GET.get("a_packname", "")
+        a_soundid = request.GET.get("a_soundid", "")
+        a_username = request.GET.get("a_username", "")
+        advanced_search_params_dict.update({  # These are stored in a dict to facilitate logging and passing to template
+            'a_tag': a_tag,
+            'a_filename': a_filename,
+            'a_description': a_description,
+            'a_packname': a_packname,
+            'a_soundid': a_soundid,
+            'a_username': a_username,
+        })
+
+        # If none is selected use all (so other filter can be appleid)
+        if a_tag or a_filename or a_description or a_packname or a_soundid or a_username != "" :
+
+            # Initialize all weights to 0
+            id_weight = 0
+            tag_weight = 0
+            description_weight = 0
+            username_weight = 0
+            pack_tokenized_weight = 0
+            original_filename_weight = 0
+
+            # Set the weights of selected checkboxes
+            if a_soundid != "":
+                id_weight = settings.DEFAULT_SEARCH_WEIGHTS['id']
+            if a_tag != "":
+                tag_weight = settings.DEFAULT_SEARCH_WEIGHTS['tag']
+            if a_description != "":
+                description_weight = settings.DEFAULT_SEARCH_WEIGHTS['description']
+            if a_username != "":
+                username_weight = settings.DEFAULT_SEARCH_WEIGHTS['username']
+            if a_packname != "":
+                pack_tokenized_weight = settings.DEFAULT_SEARCH_WEIGHTS['pack_tokenized']
+            if a_filename != "":
+                original_filename_weight = settings.DEFAULT_SEARCH_WEIGHTS['original_filename']
+
+    sort = search_prepare_sort(sort_unformatted, forms.SEARCH_SORT_OPTIONS_WEB)
+
+    logger.info(u'Search (%s)' % json.dumps({
+        'ip': get_client_ip(request),
+        'query': search_query,
+        'filter': filter_query,
+        'username': request.user.username,
+        'page': current_page,
+        'sort': sort[0],
+        'group_by_pack': grouping,
+        'advanced': json.dumps(advanced_search_params_dict) if advanced == "1" else ""
+    }))
+
+    # # we send the query parameters in the context for clustering
+    # query_params = {
```

Comment:
These are the parameters that I used to store.
They correspond to all the parameters need to perform the second step explained before (create Solr query object).